src/tokens/split_utils.c

```diff
@@ -0,0 +1,61 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   split_utils.c                                      :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: mcourtoi <mcourtoi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/03/27 20:48:05 by mcourtoi          #+#    #+#             */
+/*   Updated: 2023/03/27 21:50:34 by mcourtoi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+bool	is_white_spaces(char const c)
+{
+	return (c == ' ' || c == '\t' || c == '\n');
+}
+
+bool	check_white_spaces(char const *const str)
+{
+	return (is_white_spaces(str[0])
+		|| is_white_spaces(str[ft_strlen(str) - 1]));
+}
+
+int	strdup_or_join(t_token *const *const token, char const *const tmp)
+{
+	char	*tmp_ptr;
+
+	if (ft_strlen((*token)->str) == 0)
```

Comment:
(Again) You should check only the first character instead of calculating the whole length.